lua/starfall/libs_sv/entities.lua

```diff
@@ -347,17 +347,28 @@ function ents_methods:addCollisionListener(func)
 	if ent.SF_CollisionCallback then SF.Throw("The entity is already listening to collisions!", 2) end
 
 	local instance = SF.instance
-	ent.SF_CollisionCallback = ent:AddCallback("PhysicsCollide", function(ent, data)
-		instance:runFunction(func, SF.StructWrapper(data))
-	end)
+	if ent:GetClass() == "prop_physics" then
```

Comment:
Looks like you're using spaces instead of tabs. Also, this should be if ent:GetClass() ~= "starfall_prop"